### DISCUSSION TITLE
Should we stop --debug-private-lib in its tracks when used without a profile?

### DISCUSSION THREAD

**Discussion by ghost** [netblue30/firejail#4189] [Q&A] [answered]
When trying to check something related to `private-lib` I did a quick check of what the `--debug-private-lib` option had to show. I didn't think and run the below command with a minimal set of arguments. To my surprise it threw an `awk error` in there. Functionality didn't seem to be affected, but still, I'm not 100% sure about that.

So I'm wondering if this is expected behavior when a profile doesn't use the private-lib option to begin with or when a profile isn't provided and firejail falls back to default.profile.

```
$ firejail --debug-private-lib
Reading profile /etc/firejail/default.profile
Reading profile /etc/firejail/globals.local
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-common.local
Reading profile /etc/firejail/disable-passwdmgr.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/disable-programs.local
Warning: networking feature is disabled in Firejail configuration file

** Note: you can use --noprofile to disable default.profile **

Parent pid 227180, child pid 227181

DNS server 127.0.0.1

Warning: An abstract unix socket for session D-BUS might still be available. Use --net or remove unix from --protocol set.
Child process initialized in 764.69 ms
awk: cmd. line:1: $3=={_user_id}{print $1}
awk: cmd. line:1:     ^ syntax error
[glitsj16@lab16 Downloads]$
```

**Comment by reinerh** [comment, marked as answer]
Hm, I can't find `awk` being called anywhere from firejail.  
Is this coming from your `.bashrc` or so? Does this really only happen with `--debug-private-lib`? Or also with `--debug`, or always?

**Reply by ghost** [reply]
@reinerh Aha, nice catch! I couldn't find `awk` in the codebase either, that's why this puzzled me. Now you mentioned `.bashrc` I double-checked that and there it is indeed. Mystery solved, thanks for saving me a few hair-pulling moments :-)
